### DISCUSSION TITLE
Fastlane Testing

### DISCUSSION THREAD

**Discussion by anil150847** [fastlane/fastlane#20281] [General]
Hi I am Using fastlane in my app so that I have given some gradle tasks in the fastfile and everything is working fine but I want to know my test reports how can i find test reports in android.By using scan I am getting these 
couldn't automatically detect the project file, please provide a path:
I have given my project path but it was below error
Path must end with either .xcworkspace or .xcodeproj
and I have given those extensions also(those extension are for ios I think) can someone tell what to do to resolve this error.
Do we need to write any test cases for testing in fastlane.

**Comment by lucgrabowski** [comment]
You can use `scan` for iOS projects, not Android ones:
>how can i find test reports in android

If you want to run unit tests for Android, use `gradle` action:
https://docs.fastlane.tools/getting-started/android/running-tests/
https://docs.fastlane.tools/actions/gradle/

Please take a look at the documentation with examples if you're interested in running tests on iOS:
https://docs.fastlane.tools/actions/scan/

I don't think that I can help more. Maybe it helps or someone else will help you. Good luck.
